### ISSUE TITLE
Unable to allocate more than 32 CPU cores to WSL2

### ISSUE THREAD

**Issue by KaneGreen** [microsoft/WSL#11860]
### Windows Version

Microsoft Windows [Version 10.0.22631.3958]

### WSL Version

2.2.4.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.153.1-2

### Distro Version

Ubuntu-24.04

### Other Software

OS: Windows 11 Pro for Workstations 23H2

### Repro Steps

1. I'm using a workstation with one AMD EPYC 7K62 48-Core Processor. So, I have 48 physical cores and 96 logical cores.

2. Enable the features in Control Panel: Hyper-V, Virtual Machine Platform, Windows Hypervisor Platform and Windows Subsystem for Linux.

3. Install the latest WSL from https://github.com/microsoft/WSL/releases/latest

4. the WSL configuration in file `%USERPROFILE%/.wslconfig`
    ```
    [wsl2]
    memory=40GB
    processors=40
    defaultVhdSize=256GB
    nestedVirtualization=false
    
    [experimental]
    autoMemoryReclaim=dropcache
    sparseVhd=true
    ```
    In this configuration, I tried to allocate 40 CPU cores to WSL.

5. Install Ubuntu-24.04 from Microsoft Store.

6. Run the command `lscpu` in the WSL terminal to check how many CPU cores are available.

### Expected Behavior

I hope I can get 40 logical cores available in WSL.
```
    Thread(s) per core:  2
    Core(s) per socket:  20
    Socket(s):           1
```
or
```
    Thread(s) per core:  1
    Core(s) per socket:  40
    Socket(s):           1
```

### Actual Behavior

But I get 32 logical cores available in WSL.
```
    Thread(s) per core:  2
    Core(s) per socket:  16
    Socket(s):           1
```

-------
I also tried to specify `processors` as other values, and I found that:
* When its value is less than or equal to 32, the number of CPU cores of WSL is equal to the specified value.
* When its value is greater than 32, the number of CPU cores of WSL is always equal to 32.

In short, I can't use more than 32 cores.

**Comment by xiaokentrl** [comment]
I really love the Dynamic-Wallpaper software. If there were a Windows or Linux version, I would be willing to purchase it for a lifetime.
